cpp/include/cudf/io/parquet.hpp

```diff
@@ -487,6 +489,12 @@ class parquet_writer_options {
   size_type _max_page_size_rows = default_max_page_size_rows;
   // Maximum size of min or max values in column index
   int32_t _column_index_truncate_length = default_column_index_truncate_length;
+  // When to use dictionary encoding for data
+  dictionary_policy _dictionary_policy = dictionary_policy::ADAPTIVE;
```

Comment:
Is this in line with parquet-mr default behavior? Their docs are a bit vague on the impact of `parquet.dictionary.page.size`.

Reply:
parquet-mr does limit dictionaries to `parquet.dictionary.page.size`, but I don't believe they have any coupling between dictionary encoding and compression like this. I also think they'll still use the dictionary until it gets too big, and then do plain encoding for the rest of the chunk.